producer/src/contract/contract_test.go

```diff
@@ -136,184 +137,230 @@ func TestYieldSerialization(t *testing.T) {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 50000)
+
+	testClaim, err := MakeClaim(db, pubKey, amt+5)
 	if err == nil {
-		t.Errorf("Should have an error that resembles change")
+		t.Errorf("No change error struct returned")
+	} else {
+		if deficit, ok := err.(DeficitError); ok {
+			if deficit.Deficit != uint64(5) {
+				t.Errorf("Wrong deficit value, expected: 5, actual: %d", deficit)
+			}
+		} else {
+			t.Errorf("Different error given: %v", err)
+		}
 	}
+
 	if !bytes.Equal(testClaim.PreviousContractHash, contractHash) {
 		t.Errorf("Contract hashes do not match")
 	}
-	if testClaim.BlockIndex != 35 {
+	if testClaim.BlockIndex != blockHeight {
 		t.Errorf("Block indeces do not match")
 	}
-	if testClaim.YieldIndex != 1 {
+	if testClaim.YieldIndex != yieldIndex {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey) {
 		t.Errorf("Public Keys do not match")
 	}
-	// Case 3
-	_, shouldNotBeNil := MakeClaim("testDB.dat", testPubKey, 1)
-	if shouldNotBeNil == nil {
-		t.Errorf("Made claim on empty yield pool")
-	}
+
 }
 
-func TestMakeClaimPriorityCase1(t *testing.T) {
-	setUpDB("testDB.db")
-	defer tearDown("testDB.db")
-	testPubKey := generatePubKey()
-	testYieldFloor := MakeYield(testPubKey, 10000000)
-	testYieldCeiling := MakeYield(testPubKey, 10000010)
-	contractHashFloor := block.HashSHA256([]byte("This should not be claimed"))
-	contractHashCeiling := block.HashSHA256([]byte("This should be claimed"))
-	err := InsertYield(testYieldFloor, "testDB.dat", 35, contractHashFloor, 1)
+func TestMakeClaimMultipleYield(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := []uint64{uint64(100), uint64(200)}
+	pubKey := []ecdsa.PublicKey{generatePubKey(), generatePubKey()}
+	yield := []Yield{MakeYield(&pubKey[0], amt[0]), MakeYield(&pubKey[1], amt[1])}
+	contractHash := [][]byte{block.HashSHA256([]byte("1")), block.HashSHA256([]byte("2"))}
+	blockHeight := []uint64{uint64(35), uint64(40)}
+	yieldIndex := []uint16{uint16(1), uint16(4)}
+
+	err := InsertYield(yield[0], db, blockHeight[0], contractHash[0], yieldIndex[0])
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	err = InsertYield(testYieldFloor, "testDB.dat", 35, contractHashCeiling, 1)
+	err = InsertYield(yield[1], db, blockHeight[1], contractHash[1], yieldIndex[1])
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 10000008)
+
+	testClaim, err := MakeClaim(db, pubKey[1], uint64(50))
 	if err == nil {
-		t.Errorf("Should have an error that resembles change")
+		t.Errorf("No change error struct returned")
+	} else {
+		if change, ok := err.(ChangeError); ok {
+			if change.Change != uint64(150) {
```

Comment:
If we prioritize the `100` yield over the `200` yield then the change should be `50` not `150`.

Reply:
No, the yield of 100 corresponds to a different public key, MakeClaim should claim yield corresponding to the public key given (the yield with 200)

Reply:
Where's the test showing that `MakeClaim` prioritizes certain yield values over others?